docs/sphinx/examples/cpp/basics/noise_amplitude_damping.cpp

```diff
@@ -0,0 +1,54 @@
+// Compile and run with:
+// ```
+// nvq++ noise_amplitude_damping.cpp --target density-matrix-cpu -o dyn.x
+// --target nvidia && ./dyn.x
+// ```
+//
+// Note: You must set the target to a density matrix backend for the noise
+// to successfully impact the system.
+
+#include <cudaq.h>
+
+// CUDA Quantum supports several different models of noise. In this case,
+// we will examine the modeling of energy dissipation within our system
+// via environmental interactions. The result of this "amplitude damping"
+// is to return the qubit to the |0> state with a user-specified probability.
+
+int main() {
+
+  // We will begin by defining an empty noise model that we will add
+  // our damping channel to.
+  cudaq::noise_model noise;
+
```

Comment:
What do you think about adding `cudaq::set_random_seed(13)` to make this test repeatable? The same question applies to all the new cpp and Python tests in this review. (Similar to the spirit of PR #617)

Reply:
Thank you, and yes, that sounds reasonable to me. While we run the docs examples as part of our CI, I'm ok with a targeted subset of them being non-repeatable. As we make more progress on #608, we may want to consider updating the docs to describe to our users how to make some of the docs tests more repeatable, too, but that doesn't have to be part of this PR. Thanks!